cms/core/blocks/related.py

```diff
@@ -1,44 +1,81 @@
 from typing import TYPE_CHECKING, Any
 
+from django.forms import ValidationError
 from django.utils.text import slugify
-from django.utils.translation import gettext as _
-from wagtail.blocks import CharBlock, ListBlock
+from django.utils.translation import gettext_lazy as _
+from wagtail.blocks import CharBlock, ChoiceBlock, DateBlock, ListBlock, StructBlockValidationError
 
-from .base import LinkBlock
+from cms.core.blocks.base import LinkBlock
+from cms.core.enums import RelatedContentType
 
 if TYPE_CHECKING:
+    from django_stubs_ext import StrOrPromise
     from wagtail.blocks.list_block import ListValue
 
+    from cms.core.blocks.base import LinkBlockStructValue
 
-class RelatedContentBlock(LinkBlock):
-    """Related content block with page or link validation."""
 
+class LinkBlockWithDescription(LinkBlock):
     description = CharBlock(required=False)
 
 
+class RelatedContentBlock(LinkBlockWithDescription):
+    """Related content block with page or link validation."""
+
+    content_type = ChoiceBlock(
+        choices=RelatedContentType.choices,
+        help_text="Select the type of related content.",
+        required=False,
+    )
+    release_date = DateBlock(required=False)
+
+    def clean(self, value: "LinkBlockStructValue") -> "LinkBlockStructValue":
+        """Validate the related content based on its type."""
+        cleaned_value = super().clean(value)
+
+        errors = {}
+
+        if cleaned_value["external_url"] and not cleaned_value["content_type"]:
+            errors["content_type"] = ValidationError("You must select a content type when providing an external URL.")
```

Comment:
Q: Where did this come from? I can't see this mentioned on the card. 
I believe the aim is consistency with internal pages that have the `content_type` auto-populated, but it'd be good to confirm that we actually want it and update the card.

Reply:
This is for consistency, indeed. We can pre-populate for internal pages, but have to add a value for external URLs.

I say let's get this in and get feedback from users

Reply:
Makes sense 👍 
I read that as 
> **Then** I can _(but don't have to)_ select the type of content

It'd be good to add the word "mandatory" for the ACs relating to external links so the testers know what to expect.